### PR TITLE
Update dockerfile

### PR DESCRIPTION
**Pull request by rantan**
## Update Dockerfile

* Divide a building image and a final image to be distributed. It minimize image size.

## Update CI setting for building docker images

* Build and push docker images on Github Actions
* Build docker images on master HEAD as `edge` tag
* Build docker images on release tags like `v0.4.0`
* Make the readme document on docker hub sync with `doc/docker_image.md`

### CODE REVIEW

File: doc/docker_image.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+rpcallowip=127.0.0.1
+addseeder=static-seed.tapyrus.dev.chaintope.com
+EOS
+```
+
+#### Running tapyrus node
+
+```bash
+$ docker run -d --name 'tapyrus_node_testnet' -v $PWD/tapyrus.conf:/etc/tapyrus/tapyrus.conf -e GENESIS_BLOCK_WITH_SIG='01000000000000000000000000000000000000000000000000000000000000000000000044cc181bd0e95c5b999a13d1fc0d193fa8223af97511ad2098217555a841b3518f18ec2536f0bb9d6d4834fcc712e9563840fe9f089db9e8fe890bffb82165849f52ba5e01210366262690cbdf648132ce0c088962c6361112582364ede120f3780ab73438fc4b402b1ed9996920f57a425f6f9797557c0e73d0c9fbafdebcaa796b136e0946ffa98d928f8130b6a572f83da39530b13784eeb7007465b673aa95091619e7ee208501010000000100000000000000000000000000000000000000000000000000000000000000000000000000ffffffff0100f2052a010000002776a92231415132437447336a686f37385372457a4b6533766636647863456b4a74356e7a4188ac00000000' tapyrus/tapyrusd:latest
+
```
Comment (azuchi) [inline doc/docker_image.md:29]: `-v $PWD/tapyrus.conf:/etc/tapyrus/tapyrus.conf` seems to specify two config files, but entrypoint.sh seems to use only `conf = ${CONF_DIR}/tapyrus.conf`.

**Review by azuchi — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -1,59 +1,38 @@
...
-#
-# Example
-#      $ docker exec [Container ID] tapyrus-cli -conf=/var/lib/tapyrus/tapyrus.conf getblockchaininfo
+FROM ubuntu:18.04 as builder
 
-FROM ubuntu:18.04
-
-ENV TAPYRUS_ROOT /tapyrus-core
-ENV MAKEJOBS -j3
+ARG MAKEJOBS=3
```
[the remark this answers is not in the document]
Comment (rantan) [reply Dockerfile:3]: Ok, I will use it.

File: doc/docker_image.md (hunk 1 above)
Comment (rantan) [reply doc/docker_image.md:29]: I don't understand about the two config files. This part specifies only one config file.
`-v $PWD/tapyrus.conf:/etc/tapyrus/tapyrus.conf` tries to mount a conf file on host to container onto the path `/etc/tapyrus/tapyrus.conf`.

Comment (azuchi) [reply doc/docker_image.md:29]: Oh, I see.

**Review by azuchi — APPROVED** [review]

**Pull request merged**